File: app/src/main/java/info/nightscout/androidaps/utils/SP.java
package info.nightscout.androidaps.utils;

import android.content.SharedPreferences;
import android.preference.PreferenceManager;

import java.util.Map;

import info.nightscout.androidaps.MainApp;

/**
 * Created by mike on 17.02.2017.
 */

public class SP {
    private static SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(MainApp.instance().getApplicationContext());

    @Deprecated
    static public Map<String, ?> getAll() {
        return sharedPreferences.getAll();
    }

    @Deprecated
    static public String getString(String key, String defaultValue) {
        return sharedPreferences.getString(key, defaultValue);
    }

    @Deprecated
    static public boolean getBoolean(int resourceID, Boolean defaultValue) {
        try {
            return sharedPreferences.getBoolean(MainApp.gs(resourceID), defaultValue);
        } catch (Exception e) {
            return defaultValue;
        }
    }

    @Deprecated
    static public void putString(int resourceID, String value) {
        sharedPreferences.edit().putString(MainApp.gs(resourceID), value).apply();
    }

    @Deprecated
    static public void putString(String key, String value) {
        sharedPreferences.edit().putString(key, value).apply();
    }

    @Deprecated
    static public void remove(String key) {
        sharedPreferences.edit().remove(key).apply();
    }
}

File: app/src/main/java/info/nightscout/androidaps/utils/sharedPreferences/SP.kt
package info.nightscout.androidaps.utils.sharedPreferences

/**
 * Created by adrian on 2019-12-23.
 */

interface SP {
    fun getAll(): Map<String, *>
    fun clear()
    fun contains(key: String): Boolean
    fun contains(resourceId: Int): Boolean
    fun remove(resourceID: Int)
    fun remove(key: String)
    fun getString(resourceID: Int, defaultValue: String): String
    fun getString(key: String, defaultValue: String): String
    fun getBoolean(resourceID: Int, defaultValue: Boolean): Boolean
    fun getBoolean(key: String, defaultValue: Boolean): Boolean
    fun getDouble(resourceID: Int, defaultValue: Double): Double
    fun getDouble(key: String, defaultValue: Double): Double
    fun getInt(resourceID: Int, defaultValue: Int): Int
    fun getInt(key: String, defaultValue: Int): Int
    fun getLong(resourceID: Int, defaultValue: Long): Long
    fun getLong(key: String, defaultValue: Long): Long
    fun putBoolean(key: String, value: Boolean)
    fun putBoolean(resourceID: Int, value: Boolean)
    fun putDouble(key: String, value: Double)
    fun putLong(key: String, value: Long)
    fun putLong(resourceID: Int, value: Long)
    fun putInt(key: String, value: Int)
    fun putInt(resourceID: Int, value: Int)
    fun incInt(resourceID: Int)
    fun putString(resourceID: Int, value: String)
    fun putString(key: String, value: String)
}